File: Dynamic Programming/70. Climbing Stairs.js
/**
 * @param {number} n
 * @return {number}
 */
var climbStairs = function(n) {
    const dp = [0, 1, 2];
    for(let i = 3; i <=n; i++) {
        dp[i] = dp[i-1] + dp[i-2]; // Two base cases, so the total solution to step i will be the sum of two solutions (the step to i - 1 and step to i - 2)
    }
    return dp[n];
};

File: Golang/DynamicProgramming/70.Climbing-stairs.go
package main

func climbStairs(n int) int {
	dp := []int{0, 1, 2}
	// dp[n] = dp[n-1] + dp[n-2]
	for i := 3; i <= n; i++ {
		dp = append(dp, dp[i-1]+dp[i-2])
	}
	return dp[n]
}
